### PR TITLE
test: fix test failures in ActionEventInterceptorTest

### PR DESCRIPTION
**Pull request by weizhouapache**
This fixes test failures in some PRs against main/4.20 branch

```
Error:  Failures:
Error:    ActionEventInterceptorTest.testInterceptComplete:247
Error:    ActionEventInterceptorTest.testInterceptException:261
Error:    ActionEventInterceptorTest.testInterceptStartAsync:234 expected:<Starting VM[.  Id: 0 Name: null]> but was:<Starting VM[]>
```

### Description

This PR...

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
Not sure if it fixes the issue, let's wait for the results of github actions

**Review by DaanHoogland — APPROVED** [review]
clgtm and fixes the issue locally for me.

**Comment by weizhouapache** [comment]
> clgtm and fixes the issue locally for me.

great, thanks @DaanHoogland 

I kicked the build 3 times, all succeeded. https://github.com/apache/cloudstack/actions/runs/8734995428/job/23976479820?pr=8938

File: server/src/test/java/com/cloud/event/ActionEventInterceptorTest.java [hunk 1]
```diff
@@ -172,6 +172,7 @@ public EventVO answer(InvocationOnMock invocation) throws Throwable {
         account.setId(ACCOUNT_ID);
         user = new UserVO(1, "testuser", "password", "firstname", "lastName", "email", "timezone",
                 UUID.randomUUID().toString(), User.Source.UNKNOWN);
+        CallContext.register(user, account);
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply server/src/test/java/com/cloud/event/ActionEventInterceptorTest.java:175]: good point @vishesh92 
thanks

**Comment by weizhouapache** [comment]
marked as "ready for review" as it fixes the failure with build CI

**Pull request merged**